src/java.base/share/classes/java/io/Reader.java

```diff
@@ -382,6 +397,111 @@ public int read(char[] cbuf) throws IOException {
+     * invocations of this method will return an empty list.
+     *
+     * <p> A <i>line</i> is either a sequence of zero or more characters
+     * followed by a line terminator, or it is a sequence of one or
+     * more characters followed by the end of the stream.
+     * A line does not include the line terminator.
+     *
+     * <p> A <i>line terminator</i> is one of the following:
+     * a line feed character {@code "\n"} (U+000A),
+     * a carriage return character {@code "\r"} (U+000D),
+     * or a carriage return followed immediately by a line feed
+     * {@code "\r\n"} (U+000D U+000A).
+     *
+     * <p> The behavior for the case where the reader is
+     * <i>asynchronously closed</i>, or the thread interrupted during the
+     * read, is highly input reader specific, and therefore not specified.
+     *
+     * <p> If an I/O error occurs reading from the stream, then it
+     * may do so after some, but not all, characters have been read.
+     * Consequently the stream may not be at end of stream and may
+     * be in an inconsistent state. It is strongly recommended that the reader
+     * be promptly closed if an I/O error occurs.
+     *
+     * @apiNote
+     * This method is intended for simple cases where it is convenient
+     * to read all remaining characters in a single operation. It is not
+     * intended for reading a large number of characters, for example,
+     * greater than {@code 1G}.
+     *
+     * @return     the remaining characters as lines of text stored in an
+     *             unmodifiable {@code List} of {@code String}s in the order
+     *             they are read
+     *
+     * @throws     IOException  If an I/O error occurs
+     * @throws     OutOfMemoryError  If the number of remaining characters
+     *             exceeds the implementation limit for {@code String}.
+     *
+     * @see String#lines
+     * @see #readAllAsString
+     * @see java.nio.file.Files#readAllLines
+     *
+     * @since 25
+     */
+    public List<String> readAllLines() throws IOException {
+        return readAllCharsAsString().lines().toList();
+    }
+
+    /**
+     * Reads all remaining characters into a string. This method blocks until
+     * all remaining characters including all line separators have been read
+     * and end of stream is detected, or an exception is thrown. The resulting
+     * string will contain line separators as they appear in the stream. This
+     * method does not close the reader.
+     *
+     * <p> When this reader reaches the end of the stream, further
+     * invocations of this method will return an empty string.
+     *
+     * <p> The behavior for the case where the reader
+     * is <i>asynchronously closed</i>, or the thread interrupted during the
+     * read, is highly input reader specific, and therefore not specified.
+     *
+     * <p> If an I/O error occurs reading from the stream, then it
+     * may do so after some, but not all, characters have been read.
```

Comment:
Hello Brian, I read this sentence a couple of time but it wasn't clear to me what this meant. I am guessing that we want to state that if an I/O error occurs when some (but not all?) characters have been read then the stream may be in an inconsistent state? Should we reword this paragraph?

I suspect that sentence was motivated from similar existing text in `Files.readString(...)` and there too I find it confusing.

Reply:
Thank you Brian, the updated text for this section looks good to me. Once this PR gets integrated, I'll go through the Files.readXXX methods and file an issue to have that text simplified too.

Reply:
I see this has been changed to  "then some characters, but not all" but it doesn't flow very well. The original sentence, which was copied/modified from InputStream.readAllBytes, is much cleaner and I think I would prefer to go back to that.  Maybe Jai's issue could be address by dropping the first comma from the sentence?

Reply:
In 0880af2 reverted previous commit then removed first comma from the modified sentence.

Reply:
Hello Brian, the updated text looks OK to me. It still takes me a second read to understand what it says and I think it due to the sentence reading like an instruction to implementations that they aren't allowed to throw I/O errors before any characters have been read from the stream:

> If an I/O error occurs reading from the stream then it may do so after some ...

But there's a precedence for this text (in the API docs of `java.nio.file.Files`) and others favour the current proposed/updated form, so it's OK with me to go ahead with this text.